File: app/src/main/java/com/gals/prayertimes/db/Converters.java
package com.gals.prayertimes.db;

import androidx.room.TypeConverter;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;

public class Converters {

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Date stringToDate(String date) {
        try {
            SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
            return format.parse(date);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return null;
    }

    @TypeConverter
    static Boolean stringToBoolean(String bool) {
        return Boolean.parseBoolean(bool);
    }

}

File: app/src/main/java/com/gals/prayertimes/db/Converters.kt
package com.gals.prayertimes.db

import androidx.room.TypeConverter
import java.text.SimpleDateFormat
import java.util.Date

object Converters {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? = if (value == null) null else Date(value)

    @TypeConverter
    fun stringToDate(date: String): Date? = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").parse(date)

    @TypeConverter
    fun stringToBoolean(bool: String?): Boolean = bool.toBoolean()
}